cms/static/js/views/course_video_settings.js

```diff
@@ -0,0 +1,545 @@
+                $languagesContainer = self.$el.find('.languages-menu-container'),
+                isTurnaroundSelected = self.$el.find('#transcript-turnaround')[0].options.selectedIndex,
+                isFidelitySelected = self.$el.find('#transcript-fidelity')[0].options.selectedIndex;
+
+            // Clear error state if present any.
+            this.clearPreferanceErrorState($languagesPreferanceContainer);
+
+            $languagesContainer.empty();
+
+            if (self.selectedProvider &&
+                    ((isTurnaroundSelected > 0 && self.selectedProvider === '3PlayMedia') ||
+                    (isTurnaroundSelected  > 0 && isFidelitySelected > 0))) {
+                self.availableLanguages = self.getPlanLanguages();
+                _.each(self.activeLanguages, function(activeLanguage){
+                    // Only add if not in the list already.
+                    if (_.indexOf(self.selectedLanguages, activeLanguage) === -1) {
+                        self.selectedLanguages.push(activeLanguage);
+                        HtmlUtils.append(
+                            $languagesContainer,
+                            HtmlUtils.joinHtml(
+                                HtmlUtils.HTML('<div class="transcript-language-menu-container">'),
+                                HtmlUtils.interpolateHtml(
+                                    HtmlUtils.HTML('<span>{languageDisplayName}</span>'),
+                                    {
+                                        languageDisplayName: self.availableLanguages[activeLanguage]
+                                    }
+                                ),
+                                HtmlUtils.interpolateHtml(
+                                    HtmlUtils.HTML('<div class="remove-language-action"><button class="button-link action-remove-language" data-language-code="{languageCode}">{text}<span class="sr">{srText}</span></button></div>'),
+                                    {
+                                        languageCode: activeLanguage,
+                                        text: gettext('Remove'),
+                                        srText: gettext('Press Remove to remove language')
+                                    }
+                                ),
+                                HtmlUtils.HTML('</div>')
+                            )
+                        );
+                    }
+                });
+
+                self.addLanguageMenu();
+
+                self.$el.find('.transcript-languages-wrapper').show();
+            } else {
+                self.availableLanguages = {};
+                self.$el.find('.transcript-languages-wrapper').hide();
+            }
+        },
+
+        populatePreferences: function() {
+            this.populateProvider();
+            this.populateTurnaround();
+            this.populateFidelity();
+            this.populateLanguages();
+        },
+
+        addLanguageMenu: function() {
```

Comment:
Better if we have dropdown in template and then values should be populated on view re-render.

Reply:
I have changed this to have a dropdown in template and options updated when needed.